integration-tests/search.test.ts

```diff
@@ -118,6 +118,139 @@ describe('search', () => {
         }
     });
 
+    test('date period', async () => {
+        const documentReference = {
+            resourceType: 'DocumentReference',
+            status: 'current',
+            type: {
+                coding: [
+                    {
+                        system: 'http://loinc.org',
+                        code: '34133-9',
+                        display: 'Summary of episode note',
+                    },
+                ],
+                text: 'CCD Document',
+            },
+            subject: {
+                reference: 'Patient/example',
+                display: 'Amy Shaw',
+            },
+            content: [
+                {
+                    attachment: {
+                        contentType: 'text/plain',
+                        url: '/Binary/1-note',
+                        title: 'Uri where the data can be found: [base]/Binary/1-note',
+                    },
+                    format: {
+                        system: 'urn:oid:1.3.6.1.4.1.19376.1.2.3',
+                        code: 'urn:hl7-org:sdwg:ccda-structuredBody:2.1',
+                        display: 'Documents following C-CDA constraints using a structured body',
+                    },
+                },
+            ],
+            context: {
+                period: {
+                    start: '2010-10-10T06:00:00Z',
+                    end: '2010-10-20T06:00:00Z',
+                },
+            },
+        };
+        const testdocumentReference = (await client.post('DocumentReference', documentReference)).data;
+        await waitForResourceToBeSearchable(client, testdocumentReference);
+
+        const aFewMinutesAgo = aFewMinutesAgoAsDate();
+
+        const testsParams = [
+            { period: 'eq2010' },
+            { period: 'gt2010' },
+            { period: 'ge2010' },
+            { period: 'lt2010' },
+            { period: 'le2010' },
+            { period: 'ap2010' },
+
+            { period: 'ne2010-10-15' },
+            { period: 'lt2010-10-15' },
+            { period: 'le2010-10-15' },
+            { period: 'gt2010-10-15' },
+            { period: 'ge2010-10-15' },
+            { period: 'ap2010-10-15' },
+
+            { period: 'ne2010-10-20' },
+            { period: 'lt2010-10-20' },
+            { period: 'le2010-10-20' },
+            { period: 'gt2010-10-20' },
+            { period: 'ge2010-10-20' },
+            { period: 'ap2010-10-20' },
+
+            { period: 'ne2010-10-10' },
+            { period: 'lt2010-10-10' },
```

Comment:
These period queries are brain busters - why would this one succeed? `start: '2010-10-10T06:00:00Z` and the query is `lt2010-10-10` is it because the query is nonspecific? i.e.  `lt2010-10-10` ===`lt2010-10-10T23:59:59Z`

Reply:
All tests are against a document with period = `[2010-10-10T06:00:00Z, 2010-10-20T06:00:00Z]`

`2010-10-10` implicitly means a range of `[2010-10-10T00:00:00, 2010-10-10T23:59:59]`. The condition for `lt`( _the range below the search value intersects (i.e. overlaps) with the range of the target value_) is fulfilled since the range below `2010-10-10T23:59:59` intersects with the range of the test document period.